scene/gui/file_dialog.cpp

```diff
@@ -167,6 +167,7 @@ Vector<String> FileDialog::get_selected_files() const {
 
 void FileDialog::update_dir() {
 	dir->set_text(dir_access->get_current_dir(false));
+	dir->position_caret_at_end();
```

Comment:
Any reason not to set caret position directly from here:
 ```c++
	String dir_path = dir_access->get_current_dir(false);
	dir->set_text(dir_path);
	dir->set_caret_column(dir_path.size());
```

Adding a new function for a single use is a bit of an overkill.

Reply:
Thanks! Yeah that's probably too much; I thought that maybe a dedicated exposed method might be useful for other dialogs.

The private `_move_caret_end` is used for GUI IO apparently, which I wouldn't feel comfortable using in a programmatic way.